api/desecapi/serializers.py

```diff
@@ -284,6 +284,12 @@ def to_internal_value(self, data):
             if not isinstance(item, dict):
                 self.fail('invalid', datatype=type(item).__name__)
             s, t = self._key(item)  # subname, type
+            if not (isinstance(s, str) or s is None):
+                errors[idx].update(subname=f"Expected a string, but got {type(s).__name__}.")
+            if not (isinstance(t, str) or t is None):
+                errors[idx].update(type=f"Expected a string, but got {type(t).__name__}.")
+            if errors[idx]:
+                continue
```

Comment:
error logic is different from what we have in line 285. If `item` is not a `dict`, then `self.fail` is called. However, for the 'new' error types, `errors[idx]` is updated and the loop is continued. 

(1) Make sure the different behavior is okay/makes sense.
(2) Update comment in line 283.

Reply:
Okay, after seeing the next commit, (1) is done; but (2) should be addressed :-)

Reply:
done